test/sys/test_sockopt.rs

```diff
@@ -1063,40 +1063,54 @@ fn test_ipv6_recv_traffic_class_opts() {
     );
 }
 
+#[test]
+fn test_linger() {
+    let fd = socket(
+        AddressFamily::Inet,
+        SockType::Stream,
+        SockFlag::empty(),
+        None,
+    )
+    .unwrap();
+
+    let set_linger = libc::linger {
+        l_onoff: 1,
+        l_linger: 42,
+    };
+    setsockopt(&fd, sockopt::Linger, &set_linger).unwrap();
+
+    let get_linger = getsockopt(&fd, sockopt::Linger).unwrap();
+    assert_eq!(get_linger.l_linger, set_linger.l_linger);
+}
+
+#[cfg(apple_targets)]
+#[test]
+fn test_linger_sec() {
+    let fd = socket(
+        AddressFamily::Inet,
+        SockType::Stream,
+        SockFlag::empty(),
+        None,
+    )
+    .unwrap();
+
+    let set_linger = libc::linger {
+        l_onoff: 1,
+        l_linger: 1,
+    };
+    setsockopt(&fd, sockopt::LingerSec, &set_linger).unwrap();
+
+    let get_linger = getsockopt(&fd, sockopt::Linger).unwrap();
+    assert_eq!(get_linger.l_linger, set_linger.l_linger * 100);
+}
+
 /// Users should be able to define their own sockopts.
 mod sockopt_impl {
     use nix::sys::socket::{
         getsockopt, setsockopt, socket, AddressFamily, SockFlag, SockProtocol,
         SockType,
     };
 
-    sockopt_impl!(
-        Linger,
-        Both,
-        libc::SOL_SOCKET,
-        libc::SO_LINGER,
-        libc::linger
-    );
-    #[test]
-    fn test_linger() {
-        let fd = socket(
-            AddressFamily::Inet,
-            SockType::Stream,
-            SockFlag::empty(),
-            None,
-        )
-        .unwrap();
-
-        let set_linger = libc::linger {
-            l_onoff: 1,
-            l_linger: 42,
-        };
-        setsockopt(&fd, Linger, &set_linger).unwrap();
-
-        let get_linger = getsockopt(&fd, Linger).unwrap();
-        assert_eq!(get_linger.l_linger, set_linger.l_linger);
-    }
-
```

Comment:
Other changes look good. Could you please not remove this test from this module, it is for user-defined socket options rather than the one provided by Nix